hw_diag/utilities/miner.py

```diff
@@ -1,40 +1,30 @@
 from hm_pyhelper.constants.nebra import NEBRA_WALLET_ADDRESS
-from hm_pyhelper.miner_json_rpc import MinerClient
-from hm_pyhelper.miner_json_rpc.exceptions import MinerFailedFetchData
+from hm_pyhelper.gateway_grpc.client import GatewayClient, decode_pub_key
 from hm_pyhelper.miner_param import LOGGER
+import grpc
 
 
 def fetch_miner_data(diagnostics):
-    client = MinerClient()
-    try:
-        peerbook = client.get_peer_book()[0]
-        height = client.get_height()
-    except MinerFailedFetchData as err:
-        LOGGER.exception(err)
-        diagnostics['MC'] = "Failed to Fetch"
-        diagnostics['MD'] = "Failed to Fetch"
-        diagnostics['MH'] = None
-        diagnostics['MN'] = "Failed to Fetch"
-        diagnostics['MR'] = "Failed to Fetch"
+    with GatewayClient() as client:
```

Comment:
Somehow I missed this when I first looked. Any code that we are changing in /json should be converted into a DiagnosticReport as done in /initFile.txt. This will require a bit of refactoring but will help with stability and consistency.

Extra context: https://github.com/NebraLtd/Hotspot-Production-Tool/issues/28

Reply:
I am not fully clear on this. Does it mean:
1. I segment perform_hw_diagnostics() functionality in to several different units of Diagnostic classe
2. and then modify /json to behave the same as /initFile

Shouldn't this be a separate ticket. I have only replaced the rpc client in this ticket.

Reply:
done